### ISSUE TITLE
No dashboard in Grafana with Docker runner

### ISSUE THREAD

**Issue by dgrisham** [testground/testground#1098]
#### Version information:

I just pulled the most recent updates from `master`, rebuilt, and ran.

```
$ git log | head -n1      
commit 8b7bda0035369cb30ead74617a1e83c613f5593d

$ uname -a
Linux work 5.7.6-arch1-1 #1 SMP PREEMPT Thu, 25 Jun 2020 00:14:47 +0000 x86_64 GNU/Linux
```

#### Description:

(I'm not sure if this is a bug or my own misunderstanding of how `testground`/the underlying tools work.)

I'm getting started with testground and running a few examples (so far the `ping-pong` one, also playing with the [`bitswap-tuning`](https://github.com/ipfs/test-plans/tree/master/bitswap-tuning) test). I'm running with:

```
$ testground run single --plan=network --testcase=ping-pong --builder=docker:go --runner=local:docker --instances=2
```

which succeeds/posts data to Influx.

From what I saw, it looks like there should be a default dashboard created in Grafana based on the test output. However, when I connect to my local Grafana instance at `localhost:3000`, I don't see any data. I've tried adding an InfluxDB datasource that points to `http://localhost:8086`, no username/password, and connects to the `testground` database, but I get a `Network Error: Bad Gateway(502)` error from Grafana in response. I can, however, connect to the DB in the `testground-influxdb` container with the `influx` client, and the `testground` database exists/has series in it.

```
$ influx
Connected to http://localhost:8086 version 1.8.0
InfluxDB shell version: 1.8.0
> show databases
name: databases
name
----
testground
_internal
```

My healthcheck also looks good:

```
$ testground healthcheck --runner local:docker      
Jul  9 19:01:17.084730  INFO    using home directory: /home/grish/src/ipfs/testground
Jul  9 19:01:17.084779  INFO    .env.toml loaded from: /home/grish/src/ipfs/testground/.env.toml
Jul  9 19:01:17.084786  INFO    testground client initialized   {"addr": "localhost:8042"}

>>> Server output:

Jul  9 19:01:17.085764  INFO    checking runner: local:docker   {"req_id": "55198fee"}

>>> Result:

finished checking runner local:docker
Checks:
- local-outputs-dir: ok; directory exists.
- control-network: ok; network exists.
- local-grafana: ok; container state: running
- local-redis: ok; container state: running
- local-influxdb: ok; container state: running
- sidecar-container: ok; container state: running
No fixes applied.
```

So my questions are:

1. Should a default Grafana dashboard be created for me, or is that not what I should expect?
2. Has anyone seen the `502` error from Grafana / know how to fix that (could very well be the cause of 1)?

I've looked around the docs a fair bit, apologizes if I'm missing something obvious/stated clearly there already -- thanks!

**Comment by dgrisham** [comment]
Update: Scratch that second question, I just had to update the `Access` field in the datasource import form from `Server` to `Browser`

**Comment by nonsense** [comment]
@dgrisham thank you for this feedback, it is very valid!

At the moment:

`local:docker` runner - it just starts InfluxDB and Grafana containers, and doesn't setup automatically the `data source` for InfluxDB in Grafana unfortunately. Also doesn't setup any dashboards by default.
`cluster:k8s` runner - we install InfluxDB and Grafana helm charts, and automatically add the InfluxDB `data source` in Grafana, as well as provision a set of default dashboards.

I have been meaning to add a few tools to be able to easily set up and import dashboards into the Grafana instance (no matter if it is run locally or on a remote cluster), but these are not ready yet. It can be done with Terraform and the Grafana provider for example.

Since Testground has no way of knowing what test plans you will run, we need to come up with a way to make it easy for users/developers to load the dashboards they care about depending on what test plans they work on. For example currently we spend a lot of effort on the Oni project at https://github.com/filecoin-project/oni and we have a bunch of dashboards at https://github.com/filecoin-project/oni/tree/master/dashboards and we have to manually import these - something that I am going to automate soon.

---

Bottom line:
1. As you have noticed, this is not yet implemented.
2. In order to add InfluxDB as a data source when using `local:docker`, use the following credentials (there is no username or password):

```
host: http://testground-influxdb:8086
database: testground
```

**Comment by dgrisham** [comment]
Awesome, thanks for the rundown @nonsense! Good to know on the pending automation, I should be using testground extensively over the next few months so I'll make any contributions I can as I get more familiar!

**Comment by nonsense** [comment]
I will ping you when we have an update on this part. Feel free to send our way any other questions you might have!
